futaba/cogs/roles/core.py

```diff
@@ -205,6 +205,45 @@ async def role_remove(self, ctx, *roles: RoleConv):
         content = f"{user_discrim(ctx.author)} removed self-assignable roles: {self.str_roles(roles)}"
         self.journal.send("self/remove", ctx.guild, content, icon="role")
 
+    def get_channels_from_role(self, guild, role: RoleConv):
+        res = []
+        channel_role = self.bot.sql.roles.get_pingable_role_channels(guild)
+        for channel, role_ in channel_role:
+            if role_ == role:
+                res.append(channel)
+        return res
+
+    @role.command(name="createhelperrole", aliases=["chr"])
+    @commands.guild_only()
+    @permissions.check_mod()
+    async def helper_role_add(self, ctx, role: RoleConv, *channels: TextChannelConv):
+        helper_role = discord.utils.get(ctx.guild.roles, name=role.name + " (helper)")
+        if not helper_role:
+            helper_role = await ctx.guild.create_role(
+                name=role.name + " (helper)", colour=role.colour
+            )
+        unadded_channels = list(
+            set(channels) - set(self.get_channels_from_role(ctx.guild, helper_role))
```

Comment:
Use `frozenset()` here instead. Also the conversion to list is unnecessary.

Reply:
Unpacking a set was giving me some weird errors, will try again

Reply:
~~Premature optimization~~